plugins/builds/artifacts/get.js

```diff
@@ -16,12 +16,12 @@ module.exports = config => ({
         notes: 'Redirects to store with proper token',
         tags: ['api', 'builds', 'artifacts'],
         auth: {
-            strategies: ['session', 'token'],
-            scope: ['user']
+            strategies: ['token'],
```

Comment:
This will break current flow of opening artifacts as a link to API which in-turn redirects to store with a short lived token.

One option is to modify UI so that links are populated dynamically with token in URL or re-enable session back here.

Also one question, once a user scoped token comes does auth checks if user is actually logged in or does it ony check if token is not expired ?

Reply:
This shouldn't.  Interacting with the links should be making an AJAX call then redirecting the user.

Reply:
>  Interacting with the links should be making an AJAX call then redirecting the user.

This won't work on AJAX, UI can't redirect on AJAX and have browser provide a download dialogue, it needs JS. Simplicity of current approach is once user is logged in via session browser can handle redirection with no more JS involved after clicking the link.

If we must token, then one option is UI to download the entire file in AJAX  in memory and use JS (`URL.createObjectURL()`) to either change the current url or show file download dialog using an `a`  

Another option is for this end point to just return the store end point and use `window.open` to open a new window with the url. But that might run foul of popup settings.

I would prefer doing former approach while leaving `session` auth as it is for now and revisit this once UI has changed.

Reply:
I'll be happy to roll this back now, but we really should go with the second option.  However, instead of doing `window.open` why not `window.location`?  Since the URLs should be download only it won't _really_ cause the page to change location.

Reply:
> I'll be happy to roll this back now, but we really should go with the second option. However, instead of doing window.open why not window.location? Since the URLs should be download only it won't really cause the page to change location.

Either with `window location/open` will need change in the current UI to not follow the redirect automatically but do a `window.location` from `Location` header send back in the redirect response. But that is a change in UI, so can you put `session` back just for this route and remove it once there is a PR in UI to do this.

Reply:
Updated.